### PR TITLE
[ESIMD] Add tests for atomic operations.

### PR DESCRIPTION
**Pull request by kbobrovs**
Complementary E2E test for https://github.com/intel/llvm/pull/6629

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/lsc/atomic_smoke.cpp [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+
+std::ostream &operator<<(std::ostream &out, const Config &cfg) {
+  out << "{ thr_per_group=" << cfg.threads_per_group
+      << " n_groups=" << cfg.n_groups << " start_ind=" << cfg.start_ind
+      << " masked_lane=" << cfg.masked_lane << " repeat=" << cfg.repeat
+      << " stride=" << cfg.stride << " }";
+  return out;
+}
+
+using LSCAtomicOp = sycl::ext::intel::esimd::native::lsc::atomic_op;
```
Comment (dongkyunahn-intel) [inline SYCL/ESIMD/lsc/atomic_smoke.cpp:58]: There is no namespace named 'lsc' in intel/llvm:sycl.

Comment (dongkyunahn-intel) [reply SYCL/ESIMD/lsc/atomic_smoke.cpp:58]: Downloaded corresponding PR from intel/llvm.

**Review by v-klochkov — DISMISSED** [review]

**Review by v-klochkov — DISMISSED** [review]

File: SYCL/ESIMD/dword_atomic_smoke.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//
+//===----------------------------------------------------------------------===//
+// This test checks DWORD atomic operations.
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#define UNDEF_USE_LSC_ATOMICS
```
Comment (kbobrovs) [inline SYCL/ESIMD/dword_atomic_smoke.cpp:16]:
```suggestion
// This macro disables usage of LSC atomics in the included test.
#define UNDEF_USE_LSC_ATOMICS
```

**Review by v-klochkov — APPROVED** [review]

**Comment by kbobrovs** [comment]
Linux clean, Windows - unrelated failures due to unsupported double on TGL.

**Pull request merged**